src/functions/casing.ts

```diff
@@ -1,24 +1,24 @@
 import { Dictionary } from '@stoplight/types';
-import { ICasingOptions, IFunction } from '../types';
+import { CasingType, ICasingOptions, IFunction } from '../types';
 
-const CASES: Dictionary<RegExp, ICasingOptions['type']> = {
-  flat: /^[a-z]+$/,
-  camel: /^[a-z]+(?:[A-Z][a-z]+)*$/,
-  pascal: /^[A-Z][a-z]+(?:[A-Z][a-z]+)*$/,
-  kebab: /^[a-z]+(?:-[a-z]+)*$/,
-  cobol: /^[A-Z]+(?:-[A-Z]+)*$/,
-  snake: /^[a-z]+(?:_[a-z]+)*$/,
-  macro: /^[A-Z]+(?:_[A-Z]+)*$/,
+const CASES: Dictionary<RegExp, CasingType> = {
```

Comment:
@philsturgeon @P0lip In order for this to work, we might need https://github.com/stoplightio/types/pull/60 to be merged.

Defining the casing types as an enum allows some reflection based trickery (which makes testing the possible combinations of values easier, for instance).